### PR TITLE
use calc(), and a better devided font-family for chinese

### PR DESCRIPTION
**Pull request by Crystal-RainSlide**
13.3333 -> calc(40 / 3)
Chinese -> Chinese Simplified/Chinese Traditional

### CODE REVIEW

**Comment by tophf** [comment]
Prove it's better: show some screenshots.

File: global.css [hunk 1]
```diff
@@ -8,7 +8,7 @@ button {
   text-overflow: ellipsis;
   padding: 2px 7px;
   border: 1px solid hsl(0, 0%, 62%);
-  font: 400 13.3333px Arial;
+  font: 400 calc(40 / 3)px Arial;
```
Comment (tophf) [inline global.css]: Without a proof, I doubt there is any difference. And even if there is, we can simply use 13.3333333333333px which is the same as 40/3 due to 16-digit precision limit of IEEE754 float point numbers used internally by browsers.

**Comment by Crystal-RainSlide** [comment]
before:
before
after:
after

**Comment by Crystal-RainSlide** [comment]
Tell me if you would also like to check effect in Chinese Traditional

**Comment by tophf** [comment]
The fonts obviously look better but they are available only since Windows 8 ([1](https://docs.microsoft.com/en-us/typography/font-list/microsoft-jhenghei-ui), [2](https://docs.microsoft.com/en-us/typography/font-list/microsoft-yahei-ui)), which means you need to add a fallback font as the second item in the font list. For example our old font may be used, or a better one if you find it.

What about my comment on 13.3333 font size?

**Comment by Crystal-RainSlide** [comment]
Sorry for the day-long delay.

Windows 7 in Chinese 's `system-ui` font should just be the ideal fallback: _Microsoft YaHei_ and _Microsoft JhengHei_. While `system-ui` may not be a perfect choice for Windows 8+.

If there is no Seoge UI, the _Microsoft YaHei UI_ and _Microsoft JhengHei UI_ should be the `system-ui` of Chinese-lang Windows, but Seoge UI replace them all. Since Seoge UI is a non-CJK font, Microsoft made all the CJK fonts' availability rely on system font fallback(system wide font fallback is used for CJK long ago, but it's mainly for compatibility, not system default). Windows font fallback(and most font fallback) means use mutiple fonts one by one until fonts are used up or no more blank characters.

So if we still use `system-ui`, non-CJK characters may be shown with Seoge UI. Then the final output may contains two different font, unwanted in most case.

And, have to agree that, `system-ui` is also & always the CJK font fallback for Linux.

13.33333... That's not necessarily so important after all, you make the choice~

**Comment by tophf** [comment]
Okay, since I'm not an expert on CJK UI, I guess you know what you're talking about regarding the fonts so let's leave it as is, but please remove the font-size related changes and I'll marge the PR.

**Comment by tophf** [comment]
@eight04 maybe you can check the PR too?

**Comment by Mottie** [comment]
> I'll marge the PR

Or will you squash & marge the PR?

marge

**Comment by eight04** [comment]
It doesn't make any difference on my machine (Windows 7, Firefox, zh-TW):
image

1. There is no `Microsoft YaHei UI` and `Microsoft JhengHei UI` but `Microsoft YaHei` and `Microsoft JhengHei` on Windows 7.
2. Firefox doesn't support `system-ui`. ([MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#Browser_compatibility), [bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1226042))
3. `MingLiU` is a built-in serif font on Windows (starting from Windows 3.1 according to wiki). It doesn't make sense to specify `Arial` after `MingLiU` because it would never be used.
4. `Microsoft JhengHei` is a builtin sans-serif font on Windows (starting from Windows Vista according to wiki).

I think the correct order should be:

```css
Arial, /* use Arial for english and numbers */
'Microsoft JhengHei UI', /* fallback to JhengHei UI or JhengHei */
'Microsoft JhengHei',
system-ui, /* this only works on Chrome, and it may vary on Windows 7, 8.
              Maybe we should remove it? */
sans-serif /* use default sans-serif font, which is Arial on my Firefox
              (configurable). */
```

> the final output may contains two different font, unwanted in most case.

With the above order, the output would contain two different fonts. English/numbers would be displayed in `Arial` and others would be displayed in `Microsoft JhengHei`. If this is not intended, we can remove `Arial`.

---

BTW, why do we set `font-family` on children of `header, .actions`? Shouldn't it be configured on `body`?

**Comment by tophf** [comment]
b82b8e10d6fc59a34ad0eebb37fcb4450ff3c2ee incorporates the above, hopefully. Please test it. [Zip](https://github.com/openstyles/stylus/archive/simplified-font-css.zip).

* the font is defined on `body`
* to undo the browser's useragent stylesheet font on input/select/button we inherit it to avoid duplication of the definition or the need to use variables
* font-size is simplified to 13px - it's not like we absolutely must mimic Chrome's weird fractions, right?
* `system-ui` doesn't seem to hurt anyone AFAICT so let's keep it just in case

**Comment by eight04** [comment]
Looks good. Here are some screenshots:

<details>
<summary>Firefox</summary>

image
image
</details>
<details>
<summary>Chrome</summary>

image
image
</details>

**Comment by Crystal-RainSlide** [comment]
Hint: Firefox&Chrome's default `sans-serif` font is Microsoft JhengHei for Chinese Tranditional and Microsoft YaHei for Chinese Simplified. Since some users who are not familiar with some CJK font's issue (like: 1st version of Noto Sans CJK SC has a strange height) may set font which is problematic on the web as default `sans-serif` font. Then CJK web developers start to set `font-family` everywhere except for their self-use page.

@eight04 @tophf about `system-ui`:

> system-ui is also & always the CJK font fallback for Linux.

And maybe for macOS, Safari support it. And seems it don't have the ... Windows 8+ issue by the mixy font fallback.
Both macOS and Linux don't have a static Chinese font. And their users may set problematic font as `sans-serif`

**Pull request closed without merging**